src/main/java/com/es/phoneshop/dao/ArrayListDao.java

```diff
@@ -0,0 +1,56 @@
+package com.es.phoneshop.dao;
+
+import com.es.phoneshop.exception.ItemNotFoundException;
+import com.es.phoneshop.exception.NullValuePassedException;
+import com.es.phoneshop.model.Identifiable;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicLong;
+
+public abstract class ArrayListDao<T extends Identifiable<Long>> {
```

Comment:
Please use `Abstract` prefix for abstract classes, this will make your code easier to understand